ledger/sandbox/src/main/scala/com/digitalasset/platform/index/JdbcIndexer.scala

```diff
@@ -52,11 +52,12 @@ final abstract class Initialized extends InitStatus
 final abstract class Uninitialized extends InitStatus
 
 object JdbcIndexerFactory {
-  def apply(): JdbcIndexerFactory[Uninitialized] = new JdbcIndexerFactory[Uninitialized]()
+  def apply(namedLogger: NamedLogger): JdbcIndexerFactory[Uninitialized] =
+    new JdbcIndexerFactory[Uninitialized](namedLogger)
 }
 
-class JdbcIndexerFactory[Status <: InitStatus] private () {
-  private val logger = LoggerFactory.getLogger(classOf[JdbcIndexer])
+class JdbcIndexerFactory[Status <: InitStatus] private (protected val namedLogger: NamedLogger)
+    extends NamedLogging {
```

Comment:
@stefanobaghino-da: What's your take on `XyzLogging` traits vs. creating loggers by hand?

Reply:
Easy to go either way

Reply:
I removed the NamedLogging trait